build-aux/m4/ax_boost_base.m4

```diff
@@ -127,8 +127,14 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
       [multiarch_libsubdir="lib/${host_cpu}-${host_os}"]
     )
 
+    dnl some arches may advertise a cpu type that doesn't line up with their
+    dnl prefix's cpu type. For example, uname may report armv7l while libs are
+    dnl installed to /usr/lib/arm-linux-gnueabihf. Try getting the compiler's
+    dnl value for an extra chance of finding the correct path.
+    libsubdirs="lib/`$CXX -dumpmachine 2>/dev/null` $libsubdirs"
```

Comment:
71235217d919519422af2747d205f247bff3e4e0: I think this patch was never merged into master and the issue was fixed via 18558 (which we backported already), see 18501

Reply:
ok, I will drop this commit